src/core_plugins/kibana/public/dashboard/panel/panel.js

```diff
@@ -89,103 +74,68 @@ uiModules
        * @type {function}
        */
       onFilter: '=',
+      /**
+       * @type {Object}
+       */
       appState: '=',
     },
     link: function ($scope, element) {
       if (!$scope.panel.id || !$scope.panel.type) return;
 
-      /**
-       * Initializes the panel for the saved object.
-       * @param {{savedObj: SavedObject, editUrl: String}} savedObjectInfo
-       */
-      function initializePanel(savedObjectInfo) {
-        $scope.savedObj = savedObjectInfo.savedObj;
-        $scope.editUrl = savedObjectInfo.editUrl;
+      const savePanelState = (panel) => {
+        $scope.panel = Object.assign($scope.panel, panel);
```

Comment:
Object.assign returns its first argument, so assigning here isn't necessary, but I think it makes sense to merge instead of mutate:

```js
const updatePanelState = (changes) => {
  $scope.panel = {
    ...$scope.panel,
    ...panel
  }
  $scope.saveState();
}
```